Remove and replace `Internals::setTopContentInset()`

charliewolfe opened the pull request:
#### cdadae59887dbc28482d9cc7d1bc7eab4eb3d4b2
<pre>
Remove and replace `Internals::setTopContentInset()`
<a href="https://bugs.webkit.org/show_bug.cgi?id=277458">https://bugs.webkit.org/show_bug.cgi?id=277458</a>
<a href="https://rdar.apple.com/132944155">rdar://132944155</a>

Reviewed by Alex Christensen.

`Internals::setTopContentInset()` tries to set the top content inset on the mainframe, which can’t
happen if it’s called from a site isolated iframe. This patch moves it to `TestRunner` and replaces it
with an async version of the function.

* LayoutTests/fast/scrolling/latching/scroll-select-bottom-test.html:
* LayoutTests/fast/scrolling/mac/async-scroll-overflow-top-inset.html:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk1/fast/backgrounds/top-content-inset-fixed-attachment.html:
* LayoutTests/platform/mac-wk2/accessibility/content-inset-scrollview-frame.html:
* LayoutTests/platform/mac/fast/events/content-inset-hit-testing-in-frame.html:
* LayoutTests/platform/mac/fast/events/content-inset-hit-testing.html:
* LayoutTests/scrollingcoordinator/mac/top-content-inset-to-zero.html:
* LayoutTests/tiled-drawing/scrolling/non-fast-region/top-content-inset-header.html:
* LayoutTests/tiled-drawing/scrolling/non-fast-region/top-content-inset.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-body.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover-local-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover-local.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-local-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-local.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-positioned-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-positioned.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment.html:
* LayoutTests/tiled-drawing/visible-rect-content-inset.html:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::setTopContentInset): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetTopContentInsetForTesting):
* Source/WebKit/UIProcess/API/C/WKPagePrivate.h:
* Source/WebKit/UIProcess/WebPageProxyTesting.cpp:
(WebKit::WebPageProxyTesting::setTopContentInset):
* Source/WebKit/UIProcess/WebPageProxyTesting.h:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.cpp:
(WebKit::WebPageTesting::setTopContentInset):
* Source/WebKit/WebProcess/WebPage/WebPageTesting.h:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.messages.in:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _setTopContentInsetForTesting:]):
* Source/WebKitLegacy/mac/WebView/WebViewPrivate.h:
* Tools/DumpRenderTree/TestRunner.cpp:
(setTopContentInsetCallback):
(TestRunner::staticFunctions):
* Tools/DumpRenderTree/TestRunner.h:
* Tools/DumpRenderTree/mac/TestRunnerMac.mm:
(TestRunner::setTopContentInset):
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setTopContentInset):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):

Canonical link: <a href="https://commits.webkit.org/281770@main">https://commits.webkit.org/281770@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/23ac02e5870c761091bc6df686ead8b1b3b5964d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60869 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40228 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13445 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64800 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11416 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62999 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47904 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11691 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49212 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7920 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62903 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37454 "Found 9 new test failures: editing/deleting/smart-delete-002.html editing/deleting/smart-delete-paragraph-002.html editing/pasteboard/smart-paste-003-trailing-whitespace.html editing/pasteboard/smart-paste-004.html editing/pasteboard/smart-paste-in-text-control.html editing/pasteboard/smart-paste-paragraph-004.html editing/selection/clear-selection-crash.html editing/selection/doubleclick-whitespace-crash.html editing/selection/ios/select-text-with-long-press-actions-disabled.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52736 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30039 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34141 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9959 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10329 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55966 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10256 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66529 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4813 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10083 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56579 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4834 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52700 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56764 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3990 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9172 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36033 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37115 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38208 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36860 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Tools/DumpRenderTree/TestRunner.cpp:
```diff
@@ -1785,6 +1785,16 @@ static JSValueRef forceImmediateCompletionCallback(JSContextRef context, JSObjec
+    TestRunner* controller = static_cast<TestRunner*>(JSObjectGetPrivate(thisObject));
+    double contentInset = JSValueToNumber(context, arguments[0], exception);
+    controller->setTopContentInset(contentInset);
+    return JSValueMakeUndefined(context);
```
We may want to return JSObjectMakeDeferredPromise and resolve it immediately in this case so the await does something in DRT.  I probably should've done that with flushConsoleLogs too.

charliewolfe replied:
I can do that for both in another pr

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPageTesting.cpp:
```diff
@@ -107,6 +107,13 @@ void WebPageTesting::clearWheelEventTestMonitor()
+void WebPageTesting::setTopContentInset(float contentInset, CompletionHandler<void()>&& completionHandler)
+{
+    if (RefPtr page = m_page->corePage())
+        page->setTopContentInset(contentInset);
```
Can we instead call WebPage::setTopContentInset?

achristensen07 reviewed (approved):
Somewhat unrelated, but Messages::WebPage::SetTopContentInset isn't used on non-Cocoa platforms.  Could you add add #else to WebPage.messages.in?

charliewolfe committed:
Remove and replace `Internals::setTopContentInset()`
https://bugs.webkit.org/show_bug.cgi?id=277458
rdar://132944155

Reviewed by Alex Christensen.

`Internals::setTopContentInset()` tries to set the top content inset on the mainframe, which can’t
happen if it’s called from a site isolated iframe. This patch moves it to `TestRunner` and replaces it
with an async version of the function.

* LayoutTests/fast/scrolling/latching/scroll-select-bottom-test.html:
* LayoutTests/fast/scrolling/mac/async-scroll-overflow-top-inset.html:
* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/platform/mac-wk1/fast/backgrounds/top-content-inset-fixed-attachment.html:
* LayoutTests/platform/mac-wk2/accessibility/content-inset-scrollview-frame.html:
* LayoutTests/platform/mac/fast/events/content-inset-hit-testing-in-frame.html:
* LayoutTests/platform/mac/fast/events/content-inset-hit-testing.html:
* LayoutTests/scrollingcoordinator/mac/top-content-inset-to-zero.html:
* LayoutTests/tiled-drawing/scrolling/non-fast-region/top-content-inset-header.html:
* LayoutTests/tiled-drawing/scrolling/non-fast-region/top-content-inset.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-body.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover-local-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover-local.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-cover.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-local-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-local.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-positioned-expected.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment-positioned.html:
* LayoutTests/tiled-drawing/top-content-inset-fixed-attachment.html:
* LayoutTests/tiled-drawing/visible-rect-content-inset.html:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::setTopContentInset): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageSetTopContentInsetForTesting):
* Source/WebKit/UIProcess/API/C/WKPagePrivate.h:
* Source/WebKit/UIProcess/WebPageProxyTesting.cpp:
(WebKit::WebPageProxyTesting::setTopContentInset):
* Source/WebKit/UIProcess/WebPageProxyTesting.h:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.cpp:
(WebKit::WebPageTesting::setTopContentInset):
* Source/WebKit/WebProcess/WebPage/WebPageTesting.h:
* Source/WebKit/WebProcess/WebPage/WebPageTesting.messages.in:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _setTopContentInsetForTesting:]):
* Source/WebKitLegacy/mac/WebView/WebViewPrivate.h:
* Tools/DumpRenderTree/TestRunner.cpp:
(setTopContentInsetCallback):
(TestRunner::staticFunctions):
* Tools/DumpRenderTree/TestRunner.h:
* Tools/DumpRenderTree/mac/TestRunnerMac.mm:
(TestRunner::setTopContentInset):
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::setTopContentInset):
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):

Canonical link: https://commits.webkit.org/281770@main
